test/OpenTelemetry.Shims.OpenTracing.Tests/SpanBuilderShimTests.cs

```diff
@@ -251,12 +261,13 @@ public void WithTag_ValueIsNullStringValue()
         var tracer = TracerProvider.Default.GetTracer(TracerName);
         var shim = new SpanBuilderShim(tracer, "foo");
 
-        shim.WithTag("foo", null);
+        shim.WithTag("foo", null!);
```

Comment:
It might be better to keep the string parameter non-nullable as a safeguard. This approach ensures that null values are handled consistently and prevents unexpected behavior. By coalescing to string.Empty, we're enforcing a clear, predictable contract, which reduces the chances of errors and confusion, rather than allowing nulls to propagate.

I also updated the comment to reflect this approach 39adc122db7087db54d93aa6c399403d3aa91df3